### PR TITLE
Add GroqAIModel support

### PR DESCRIPTION
**Pull request by innovatorved**
This pull request introduces support for the GroqAI model in the `ai_hawk` project. The changes include adding a new class for the GroqAI model and updating relevant methods to handle this new model type.

Key changes include:

### Adding GroqAI model support:

* [`src/ai_hawk/llm/llm_manager.py`](diffhunk://#diff-28b4640676a995e39c3609588b59c94e3945ad30b6939cf4c9fd773a6007b839R32-R41): Added `GroqAIModel` class to handle interactions with the GroqAI API.
* [`src/ai_hawk/llm/llm_manager.py`](diffhunk://#diff-28b4640676a995e39c3609588b59c94e3945ad30b6939cf4c9fd773a6007b839R137-R138): Updated `_create_model` method to include a case for creating a `GroqAIModel` instance.

### Updating logger classes to support GroqAI model:

* [`src/ai_hawk/llm/llm_manager.py`](diffhunk://#diff-28b4640676a995e39c3609588b59c94e3945ad30b6939cf4c9fd773a6007b839L136-R148): Modified `LLMLogger` and `LoggerChatModel` constructors to accept `GroqAIModel` instances. [[1]](diffhunk://#diff-28b4640676a995e39c3609588b59c94e3945ad30b6939cf4c9fd773a6007b839L136-R148) [[2]](diffhunk://#diff-28b4640676a995e39c3609588b59c94e3945ad30b6939cf4c9fd773a6007b839L244-R256)

### CODE REVIEW

**Comment by surapuramakhil** [comment]
@innovatorved can you assign feature request

**Comment by innovatorved** [comment]
@surapuramakhil I hadn't reviewed the request initially; I made the changes for my own use but realized they could benefit others as well. 
Since Groq offers generous free limits, I thought this could be helpful.

File: src/ai_hawk/llm/llm_manager.py [hunk 1]
```diff
@@ -133,7 +145,7 @@ def invoke(self, prompt: str) -> str:
 
 class LLMLogger:
 
-    def __init__(self, llm: Union[OpenAIModel, OllamaModel, ClaudeModel, GeminiModel]):
+    def __init__(self, llm: Union[OpenAIModel, OllamaModel, ClaudeModel, GeminiModel, GroqAIModel]):
```
Comment (surapuramakhil) [inline src/ai_hawk/llm/llm_manager.py:148]: can you update this to base class AIModel and see if that works

**Review by surapuramakhil — CHANGES_REQUESTED** [review]
update Readme

**Comment by innovatorved** [comment]
Updated README as suggested.

File: src/ai_hawk/llm/llm_manager.py (hunk 1 above)
Comment (innovatorved) [reply src/ai_hawk/llm/llm_manager.py:148]: Could you clarify? From my side, the current setup seems to be working fine.

Comment (surapuramakhil) [reply src/ai_hawk/llm/llm_manager.py]: I have made a commit updating this. can you test your branch once

Comment (innovatorved) [reply src/ai_hawk/llm/llm_manager.py]: Yeah, its working fine !

**Review by surapuramakhil — APPROVED** [review]

**Comment by surapuramakhil** [comment]
@innovatorved request to resolve conflits.

**Comment by innovatorved** [comment]
@surapuramakhil kindly check and merge

File: requirements.txt [hunk 2]
```diff
@@ -6,6 +6,7 @@ jsonschema==4.23.0
 jsonschema-specifications==2023.12.1
 langchain==0.2.11
 langchain-anthropic
+langchain-groq
```
Comment (surapuramakhil) [inline requirements.txt:9]: add version by doing pip freeze

**Review by surapuramakhil — CHANGES_REQUESTED** [review]
We have seen that `pip install` is breaking on new setups. Having a working version is crucial

File: requirements.txt (hunk 2 above)
Comment (surapuramakhil) [reply requirements.txt]: CI Test passed

**Review by surapuramakhil — APPROVED** [review]

**Pull request merged**